presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -189,6 +189,9 @@ public static long unscaledDecimalToUnscaledLongUnsafe(Slice decimal)
         return negative ? -lo : lo;
     }
 
+    /*
```

Comment:
Sometimes it might be more efficient to do this check inside multiply/add function, because we don't have to unpack integers from Slice. But whatever, future improvement.